packages/firestore/exp/src/api/database.ts

```diff
@@ -234,64 +107,98 @@ export function enableIndexedDbPersistence(
   firestore: firestore.FirebaseFirestore
 ): Promise<void> {
   const firestoreImpl = cast(firestore, Firestore);
-  return firestoreImpl._enablePersistence(
+  const settings = firestoreImpl._getSettings();
+  return initializeFirestoreClient(
+    firestoreImpl,
     new IndexedDbComponentProvider(),
-    /*synchronizeTabs=*/ false
+    {
+      durable: true,
+      synchronizeTabs: false,
+      cacheSizeBytes:
+        settings.cacheSizeBytes || LruParams.DEFAULT_CACHE_SIZE_BYTES,
+      forceOwningTab: false
+    }
   );
 }
 
 export function enableMultiTabIndexedDbPersistence(
   firestore: firestore.FirebaseFirestore
 ): Promise<void> {
   const firestoreImpl = cast(firestore, Firestore);
-  return firestoreImpl._enablePersistence(
+  const settings = firestoreImpl._getSettings();
+  return initializeFirestoreClient(
+    firestoreImpl,
     new MultiTabIndexedDbComponentProvider(),
-    /*synchronizeTabs=*/ false
+    {
+      durable: true,
+      synchronizeTabs: true,
+      cacheSizeBytes:
+        settings.cacheSizeBytes || LruParams.DEFAULT_CACHE_SIZE_BYTES,
+      forceOwningTab: false
+    }
   );
 }
 
 export function clearIndexedDbPersistence(
   firestore: firestore.FirebaseFirestore
 ): Promise<void> {
   const firestoreImpl = cast(firestore, Firestore);
-  return firestoreImpl._clearPersistence((databaseId, persistenceKey) => {
-    return indexedDbClearPersistence(
-      indexedDbStoragePrefix(databaseId, persistenceKey)
+  if (firestoreImpl._initialized && !firestoreImpl._terminated) {
+    throw new FirestoreError(
+      Code.FAILED_PRECONDITION,
+      'Persistence can only be cleared before a Firestore instance is ' +
+        'initialized or after it is terminated.'
     );
+  }
```

Comment:
Suggestion: ConsiderPut this into an assert method like `assertIsNotInitialized()`.

Reply:
Kept as is for now since this is the only callsite and the only method that I am aware of that has this contract (plus I would have to come up with a good name - assertIsNotInitialized is only 90% correct...)

Reply:
Okay.